ui/raidboss/data/06-ew/raid/p7s.ts

```diff
@@ -134,19 +138,47 @@ const triggerSet: TriggerSet<Data> = {
       netRegex: NetRegexes.gainsEffect({ effectId: ['CEC', 'D45'] }),
       condition: Conditions.targetIsYou(),
       durationSeconds: 20,
-      response: (_data, matches, output) => {
+      response: (data, matches, output) => {
         // cactbot-builtin-response
         output.responseOutputStrings = {
           stackThenSpread: Outputs.stackThenSpread,
           spreadThenStack: Outputs.spreadThenStack,
         };
 
+        // Strore debuff for reminders
+        data.previousBondsDebuff = matches.effectId;
+
         const longTimer = parseFloat(matches.duration) > 9;
         if (longTimer)
           return { infoText: output.stackThenSpread!() };
         return { infoText: output.spreadThenStack!() };
       },
     },
+    {
+      id: 'P7S Inviolate Bonds Reminders',
+      // First trigger is ~4s after debuffs callout
+      // These happen 6s before cast
+      type: 'HeadMarker',
+      netRegex: NetRegexes.headMarker({}),
+      suppressSeconds: 1,
+      infoText: (data, matches, output) => {
+        const correctedMatch = getHeadmarkerId(data, matches);
+        if (correctedMatch === '00A6' && data.purgationDebuffCount === 0) {
+          switch (data.previousBondsDebuff) {
+            case 'CEC':
+              data.previousBondsDebuff = 'D45';
```

Comment:
Why do you need to set this again here?

Rather than using effect ids, could `P7S Inviolate Bonds` just set some `data.finalBondsDebuff = 'spread';` and then this trigger could read it? That seems a little bit more straight forward than remembering the previous as an effect id and then having to switch it here.

Reply:
I think it's leftover from an original thought to use this same variable for purgation, but I instead used effects there since it had what I needed.

Reply:
The trigger is used for both the first debuff and the second debuff, so it needs to check which output to use.